NestJS/src/Messages/message.listener.ts

```diff
@@ -4,30 +4,75 @@ import { Controller } from '@nestjs/common';
 import * as Amqp from "amqp-ts";
 import { RecipientService } from 'Recipients/recipient.service';
 
-var connection = new Amqp.Connection("amqp://localhost");
-var exchange = connection.declareExchange("test3", 'direct', { durable: false });
-var queue = connection.declareQueue('');
+const connection = new Amqp.Connection("amqp://localhost");
+const exchange = connection.declareExchange("js-backend", 'direct', { durable: false });
+const queue = connection.declareQueue('message', {durable: false});
 
-@Controller('message')
+@Controller('messages')
 export class MessageListener {
+
     constructor(private readonly messageService: MessageService,
     private readonly recipientService: RecipientService){
-      queue.bind(exchange, 'message');
+      this.listenQueue();
+  }
+
+  private listenQueue() {
+    queue.bind(exchange, 'message');
       queue.activateConsumer((message) => {
         var msg = message.getContent()
+        console.log(msg)
         var data = JSON.parse(msg)
-        qs.sendMessageH(data)
-        console.log("Message is send!")
-        getMsgId(data)
+        this.takeAction(data);
+    }, {noAck: true})
+  }
 
-        async function getMsgId(data){
-          const a = await messageService.create(data)
-          data.recipients.map((rec) => rec.message = a)
+  private takeAction(msg) {
+     let res;
+    switch(msg.task){
+      case "send":
+        res = this.sendEmail(msg)
+        break;
+      case "get":
+        res = this.getMessages(msg)
+        break;
+    }
+    return res 
+  }
 
-          return recipientService.create(data.recipients)
-        }
+  async getMessages(data){
+    const msgs = await this.messageService.findByRecipient(data.recipient)
+    const response = JSON.stringify(msgs)
+   this.sendResponse(response)
+    return response
+  }
 
-    }, {noAck: true})
+  sendResponse(res) {
+    connection.declareQueue("message-response")
+    connection.completeConfiguration().then(() => {
+      var msg2 = new Amqp.Message(res);
+      exchange.send(msg2);
+      console.log(' [x] Sent message-response  \'' + msg2.getContent() + '\'');
+    });
+  }
+
+  async sendEmail(data){
+      const someFunc = async (data) => {
+        const promises = data.recipients.map((element) => qs.sendMessageH(data, element));
+        try {
+            const results = await Promise.all(promises);
+            this.getMsgId(data)
+            return results
+        } catch (err) {
+            return console.error(err)
+        }
+      } 
+      const results = await someFunc(data)
+      this.sendResponse(results)
+  }
```

Comment:
this can be simplified to:
```
async sendEmail(data){
    const results = await Promise.all(data.recipients.map((element) => qs.sendMessageH(data, element)));
    try {
      await this.getMsgId(data);  
    } catch (e) {
      console.log(e)
    }
    this.sendResponse(results)
  }
```

Reply:
Yes, Thank you!